src/agent.spec.ts

```diff
@@ -1,74 +1,173 @@
 import {
-  FindingType,
-  FindingSeverity,
   Finding,
-  HandleTransaction,
+  FindingSeverity,
+  FindingType,
   createTransactionEvent,
-  ethers,
+  HandleTransaction,
+  TransactionEvent,
+  
 } from "forta-agent";
-import agent, {
-  ERC20_TRANSFER_EVENT,
-  TETHER_ADDRESS,
-  TETHER_DECIMALS,
-} from "./agent";
+import { TestTransactionEvent } from "forta-agent-tools/lib/test";
+import provideHandleTransaction from "./agent";
+import { Interface } from "@ethersproject/abi";
+import { FORTA_BOT_REGISTRY,BOT_DEPLOYER_ADDRESS, NEW_AGENT_FUNCTION_SIGNATURE} from "./constants";
+import { BigNumber } from "ethers";
 
-describe("high tether transfer agent", () => {
-  let handleTransaction: HandleTransaction;
-  const mockTxEvent = createTransactionEvent({} as any);
+//mock txs data:
 
+const mockTxEventOne = {
+  agentId:"123456",
+  owner: BOT_DEPLOYER_ADDRESS,
+  chainIds:["56"],
+  metadata: "Dummy info 1",
+}
+
+const mockTxEventTwo = {
+  agentId:"654321",
+  owner: BOT_DEPLOYER_ADDRESS,
+  chainIds:["56"],
+  metadata: "Dummy info 2",
+}
+
+
+describe("bot deployment agent", () => {
+  let handleTransaction: HandleTransaction;
+  let mockTxEvent = createTransactionEvent({} as any);
+  let findings =
   beforeAll(() => {
-    handleTransaction = agent.handleTransaction;
+    handleTransaction = provideHandleTransaction(FORTA_BOT_REGISTRY,BOT_DEPLOYER_ADDRESS,NEW_AGENT_FUNCTION_SIGNATURE);
   });
 
-  describe("handleTransaction", () => {
-    it("returns empty findings if there are no Tether transfers", async () => {
-      mockTxEvent.filterLog = jest.fn().mockReturnValue([]);
-
-      const findings = await handleTransaction(mockTxEvent);
+  const proxyInterface = new Interface([NEW_AGENT_FUNCTION_SIGNATURE]);
 
+  describe("handleTransaction", () => {
+    //no bot deployments case
+    it("returns empty findings if there are no bot deployments", async () => {
+      mockTxEvent = new TestTransactionEvent();
+      findings = await handleTransaction(mockTxEvent);
       expect(findings).toStrictEqual([]);
-      expect(mockTxEvent.filterLog).toHaveBeenCalledTimes(1);
-      expect(mockTxEvent.filterLog).toHaveBeenCalledWith(
-        ERC20_TRANSFER_EVENT,
-        TETHER_ADDRESS
-      );
     });
+    //from != to deployer address case
+    it("returns empty findings if the deployer is NOT Nethermind", async()=>{
+      mockTxEvent = new TestTransactionEvent()
+      .setFrom("0x12345") //set the from value to a dummy value thats not the nethermind contract address
```

Comment:
Look into the `createAddress` method from the `forta-agent-tools`. Example [here](https://github.com/NethermindEth/Forta-Agents/blob/main/DYDX-Bots/Roles-Monitor/src/agent.spec.ts#L27-L34).